src/CommandLine/Core/TokenPartitioner.cs

```diff
@@ -23,11 +24,11 @@ public static
             var scalars = Scalar.Partition(tokenList, typeLookup).Memorize();
             var sequences = Sequence.Partition(tokenList, typeLookup).Memorize();
             var nonOptions = tokenList
-                .Where(t => !switches.Contains(t))
-                .Where(t => !scalars.Contains(t))
-                .Where(t => !sequences.Contains(t)).Memorize();
+                .Where(t => !switches.Contains(t, ReferenceEqualityComparer.Default))
+                .Where(t => !scalars.Contains(t, ReferenceEqualityComparer.Default))
+                .Where(t => !sequences.Contains(t, ReferenceEqualityComparer.Default)).Memorize();
             var values = nonOptions.Where(v => v.IsValue()).Memorize();
-            var errors = nonOptions.Except(values).Memorize();
+            var errors = nonOptions.Except(values, ReferenceEqualityComparer.Default).Cast<Token>().Memorize();

```

Comment:
But reference comparer doesn't have to be generic because it compares references anyway. I used [this implementation from stackoverflow](http://stackoverflow.com/a/35520207). As author says it's using contravariance feature so we can cast comparer to any type of IEqualityComparer<T>.